tests/test_local_move.py

```diff
@@ -0,0 +1,183 @@
+from jax import config, grad, jit
+from jax import numpy as jnp
+from jax import vmap
+
+config.update("jax_enable_x64", True)
+
+from functools import partial
+
+import numpy as np
+import pytest
+
+from timemachine.integrator import VelocityVerletIntegrator
+from timemachine.md.local_resampling import local_resampling_move
+from timemachine.potentials.jax_utils import delta_r
+
+
+def make_hmc_mover(x, logpdf_fxn, dt=0.1, n_steps=100):
+    masses = np.ones(len(x))
+
+    def force_fxn(x):
+        return -grad(logpdf_fxn)(x)
+
+    integrator = VelocityVerletIntegrator(force_fxn, masses=masses, dt=dt)
+
+    def augmented_logpdf(x, v):
+        return logpdf_fxn(x) - np.sum(0.5 * masses[:, np.newaxis] * v ** 2)
+
+    @jit
+    def augmented_proposal(x0, v0):
+        logp_before = augmented_logpdf(x0, v0)
+        x1, v1 = integrator._update_via_fori_loop(x0, v0, n_steps=n_steps)
+        logp_after = augmented_logpdf(x1, v1)
+
+        log_accept_prob = jnp.clip(jnp.nan_to_num(logp_after - logp_before, nan=-np.inf), a_max=0.0)
+
+        return (x1, v1), log_accept_prob
+
+    def hmc_move(x0):
+        v0 = np.random.randn(*x.shape)
+        (x1, v1), log_accept_prob = augmented_proposal(x0, v0)
+
+        if np.random.rand() < jnp.exp(log_accept_prob):
+            return x1, log_accept_prob
+        else:
+            return x0, log_accept_prob
+
+    return hmc_move
+
+
+def expect_no_drift(x0, move_fxn, observable_fxn, n_local_resampling_iterations=100):
+    traj = [jnp.array(x0)]
+    aux_traj = []
+
+    for _ in range(n_local_resampling_iterations):
+        updated, aux = move_fxn(traj[-1])
+
+        traj.append(updated)
+        aux_traj.append(aux)
+
+    expected_selection_fraction_traj = np.array([observable_fxn(x) for x in traj])
+
+    # TODO: don't hard-code T, thresholds, etc.
+    T = 10
+    assert n_local_resampling_iterations > 2 * T
+    avg_at_start = np.mean(expected_selection_fraction_traj[:T])
+    avg_at_end = np.mean(expected_selection_fraction_traj[-T:])
+
+    deviated_by_50percent_or_more = (avg_at_start / avg_at_end) <= 0.5 or (avg_at_start / avg_at_end) >= 1.5
```

Comment:
The ratios here seem flipped if we mean deviated from start to end by 50% or more?

Reply:
Good call -- fixed in https://github.com/proteneer/timemachine/pull/785/commits/5aceb6884604415c9e88ef2d17e06aeda4f34e60